src/rpc/net.cpp

```diff
@@ -215,14 +214,19 @@ UniValue addnode(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
     std::string strNode = request.params[0].get_str();
+    const bool privileged = request.params[2].isNull() ? true : request.params[2].get_bool();
```

Comment:
WDYT moving this to `onetry` if block below.

Reply:
It's needed for all cases.